AwfulCore/Sources/AwfulCore/Networking/ForumsClient.swift

```diff
@@ -220,7 +220,10 @@ public final class ForumsClient {
             .map(on: backgroundContext) { result, context -> [NSManagedObjectID] in
                 let threads = try result.upsert(into: context)
                 _ = try result.upsertAnnouncements(into: context)
-
+                
+                // we only know if the forum allows posting while scraping forumdisplay.php
+                forum.canPost = result.canPostNewThread
```

Comment:
Good catch!

Reply:
Turns out, not 100% true lol
I've added an additional way to set the forum.canPost value in other files. It was possible to bookmark a goldmined post via the site and visit the bookmark in the app, avoiding the thread list (and this function) altogether.

Now, in addition to this, the forum group ID is checked. If it's 49 (Archives) then set canPost to false.

Reply:
I often get confused with "posting", so just to clarify: the Forums code usually separates "posting" (making new threads) from "replying" (adding to an existing thread). Whereas Forums users never say "reply", and an out-of-context "post" usually means "reply" in this parlance.

So we should assume the Forums code has different permissions for "post (new threads)" and "reply (to an existing thread)". And ideally we figure it out by looking for the availability of a "Post (new thread)" or "Reply" button on the appropriate page.

Reply:
Yeah the issue I found was that threads aren't closed in the goldmine. The html still shows the regular reply button, tho on the site it's hidden with css.

So there's not really a way to know without checking the forum group? I will revisit anyway but that's my thinking

Reply:
Yep I think that's the best we can do until we add (or write :getin:) a CSS parser. Let's go for it here, but let's not turn off the "reply to thread" compose button based on it.